### PR TITLE
added frame type for path step

### PR DESCRIPTION
**Pull request by nicovanbentum**

### CODE REVIEW

File: code/headers/frame_types.hpp [hunk 1]
```diff
@@ -262,11 +263,26 @@ namespace r2d2 {
         bool brake;
     };
 
+    /**
+     * Struct thats used as a "step" in an A-star algorithm path.
+     * 
+     * Navigation wiki:
+     * https://github.com/R2D2-2019/R2D2-2019/wiki/Navigation
+     */
+    struct frame_path_step_s {
```
Comment (itzandroidtab) [inline code/headers/frame_types.hpp:376]: Can you change the order of the variables to have the lowest size at the bottem. And add a 

```cpp
#pragma pack(1)
```

Before the struct

**Comment by itzandroidtab** [comment]
What kind of step is this? Does it give a coordinate to to or does it provide a offset from the current location. What does path_id do? What does the step_id do? Doesnt a location have a north_south/ east_west direction? And a altitude?

File: code/headers/frame_types.hpp (hunk 1 above)
Comment (itzandroidtab) [reply code/headers/frame_types.hpp:376]: This removes the padding from the struct and saves at least 1 byte

**Comment by itzandroidtab** [comment]
Maybe a coordinate struct in this struct is a better idea

File: code/headers/frame_types.hpp (hunk 1 above)
Comment (nicovanbentum) [reply code/headers/frame_types.hpp:376]: both fixed in 2nd last commit. Regarding coordinates, we realize we're not inline with Location detection/Mapping but we have discussed this with Swarm Management and came to the conclusion  that it's up to Location Detection to fix their stuff.

**Comment by LRstudentHU** [comment]
Please update from master again

**Review by LRstudentHU — APPROVED** [review]

**Pull request merged**